core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -137,6 +153,73 @@ class AdminClient(val time: Time,
       broker -> Try[NodeApiVersions](new NodeApiVersions(getApiVersions(broker).asJava))
     }.toMap
 
+
+  /*
+   * Remove all the messages whose offset is smaller than the given offset of the corresponding partition
+   *
+   * PurgeDataResult contains either lowWatermark of the partition or exception. We list the possible exception
+   * and their interpretations below:
+   *
+   * - DisconnectException if leader node of the partition is not available. Need retry by user.
```

Comment:
Yes we can, if we ask user to specify retries num and retry back off. Current implementation assumes retries = 1. This seems like an optimization which may be added later?